SCClassLibrary/Common/Core/CondVar.sc

```diff
@@ -79,11 +79,14 @@ CondVar {
 
 			didNotTimeout = false;
 
-			// wokenThread must be non-nil. If our waiting thread is signaled before
-			// this timeout fires, the timeout routine will be stopped after prWait
-			// below and we won't get here.
+			// If our waiting thread is signaled before this timeout fires
+			// the timeout routine will be stopped after prWait below
+			// and we won't get here.
 			wokenThread = this.prRemoveWaitingThread(waitingThread);
-			this.prWakeThread(wokenThread);
+			// if wokenThread is nil, we asssume a signal happened
+			// immediately prior to timeout and the thread is already awake
+			// no call to prWakeThread happens in this case
+			wokenThread !? { this.prWakeThread(wokenThread) };
```

Comment:
I'll offer this only as a comment, not a "requested change," but... as I posted recently on the forum, I find more and more that it improves readability and future maintainability by spelling out the logic rather than using funky operators. In my opinion, I'd like to see funky operators used less and named methods more.

If I were making this change, I would write:

```
if(wokenThread.notNil) { this.prWakeThread(wokenThread) };
```

Incidentally:

```
{ a !? { b } }.def.dumpByteCodes
BYTECODES: (7)
  0   12       PushInstVar 'a'
  1   8F 1B 00 01 ControlOpcode 1  (5)
  5   13       PushInstVar 'b'
  6   F2       BlockReturn

{ if(a.notNil) { b } }.def.dumpByteCodes
BYTECODES: (7)
  0   12       PushInstVar 'a'
  1   8F 1B 00 01 ControlOpcode 1  (5)
  5   13       PushInstVar 'b'
  6   F2       BlockReturn
```

So the more explicit style, in this case, has no performance cost (identical byte codes), but would have benefits for future readers -- not much of a downside.

Reply:
I'll re-write it as an if statement for clarity :+1: